src/CMakeLists.txt

```diff
@@ -200,8 +200,20 @@ endif()
 # Python stuff
 find_package(PythonInterp 2 REQUIRED)
 find_package(PythonLibs 2 REQUIRED)
-find_package(Python3Interp 3 REQUIRED)
-find_package(Python3Libs 3 REQUIRED)
+
+option(WITH_PY3 "build python3 bindings" "CHECK")
+if(WITH_PY3 MATCHES "check|CHECK")
+  find_package(Python3Interp 3 QUIET)
+  find_package(Python3Libs 3 QUIET)
+  if(PYTHON3INTERP_FOUND AND PYTHON3LIBS_FOUND)
+    set(WITH_PY3 ON)
+  else()
+    set(WITH_PY3 OFF)
+  endif()
+elseif(WITH_PY3)
+  find_package(Python3Interp 3 REQUIRED)
+  find_package(Python3Libs 3 REQUIRED)
+endif()
```

Comment:
i didn't check if other included `CMakeLists.txt` scripts are using the variables exposed by `find_package(Python3Interp 3)` and `find_package(Python3Libs)` or not. so not sure about this.